lib/std/mem.zig

```diff
@@ -196,22 +196,54 @@ test "Allocator.resize" {
 /// `copyForwards` if they do.
 pub const copy = copyForwards;
 
+/// Returns true if the slices overlap
+pub fn slicesOverlap(comptime T: type, slice_a: []const T, slice_b: []const T) bool {
+    // NOTE: We can't currently determine if ranges overlap at comptime (as the
+    // currently language doesn't expose the information needed to do that).
+    const a_begin = @intFromPtr(slice_a.ptr);
+    const b_begin = @intFromPtr(slice_b.ptr);
+    const a_end = a_begin + (@sizeOf(T) * slice_a.len);
+    const b_end = b_begin + (@sizeOf(T) * slice_b.len);
+
+    if (a_begin > b_begin) {
+        // a starts after b, so confirm b_end doesn't overlap a_begin!
+        return b_end <= a_begin;
+    } else if (a_begin < b_begin) {
+        // b starts after a, so confirm a_end doesn't overlap b_begin!
+        return a_end <= b_begin;
+    }
```

Comment:
The new `copyForwards`, `copyBackwards` enforce pointer ordering if we return `true` here.
Since there's only danger of overwriting if `a_end != b_begin`, I think the `return`-ed condition should be `<` instead of `<=` - I think that's the source of the current CI errors.

Reply:
yeah! independetntly noticed that when reviewing things myself, cheers for that! 👍